packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -242,24 +247,58 @@ export default {
       fields: fieldsToSelect,
     };
 
+    const permissionChecker = getService('permission-checker').create({
+      userAbility,
+      model: targetedModel.uid,
+    });
+    const permissionQuery = await permissionChecker.sanitizedQuery.read(queryParams);
+
     if (isAnyToMany(attribute)) {
-      const res = await strapi.entityService.loadPages(
-        model,
-        { id },
-        targetField,
-        {
-          ...queryParams,
-          ordering: 'desc',
-        } as any,
-        {
-          page: ctx.request.query.page,
-          pageSize: ctx.request.query.pageSize,
-        }
+      const [resWithOnlyIds, res] = await Promise.all([
+        strapi.entityService.loadPages(
+          model,
+          { id },
+          targetField,
+          {
+            fields: ['id'],
+          } as any,
+          {
+            page: ctx.request.query.page,
+            pageSize: ctx.request.query.pageSize,
+          }
+        ),
+        strapi.entityService.loadPages(
+          model,
+          { id },
+          targetField,
+          {
+            ...permissionQuery,
+            ordering: 'desc',
+          } as any,
+          {
+            page: ctx.request.query.page,
+            pageSize: ctx.request.query.pageSize,
+          }
+        ),
+      ]);
+
+      // Filter out the results that don't exist in resWithIds to preserve correct pagination
+      const results = res.results.filter((item: any) =>
+        resWithOnlyIds.results.some((s: any) => item.id === s.id)
       );
+      res.results = uniqBy('id', concat(results, resWithOnlyIds.results));
 
       ctx.body = res;
     } else {
-      const result = await strapi.entityService.load(model, { id }, targetField, queryParams);
+      const [resWithOnlyId, res] = await Promise.all([
+        await strapi.entityService.load(model, { id }, targetField, {
+          fields: ['id'],
+        }),
```

Comment:
I believe the await is not necessary here, as the promise will be awaited by the promise all